lang/cs/Org.Apache.REEF.Wake.Tests/TcpConnectionRetryTest.cs

```diff
@@ -55,23 +55,22 @@ public void TestConnectionRetries()
             try
             {
                 tmp.Connect(remoteEndpoint);
-                Assert.False(true);
+                Assert.False(true, "The connection is supposed to be unsuccessful since server is not present.");
             }
-            catch
+            catch (Exception e)
             {
-                memStream.Position = 0;
-                using (var reader = new StreamReader(memStream))
+                var exception = e as TcpClientConnectionException;
+                if (exception != null)
                 {
-                    string line;
-                    int counter = 0;
-                    while ((line = reader.ReadLine()) != null)
-                    {
-                        if (line.Contains(message))
-                        {
-                            counter++;
-                        }
-                    }
-                    Assert.Equal(counter, retryCount);
+                    bool areEqual = exception.RetriesDone == retryCount;
```

Comment:
Assert.Equal doesn't allow to add error message. It's not very informative to get a failure "Expected: 0, Actual: 5", especially when the arguments in Assert.Equal are accidentally switched and 5 is expected but 0 is expected.